src/category_theory/sites/sieves.lean

```diff
@@ -41,6 +41,17 @@ namespace presieve
 
 instance : inhabited (presieve X) := ⟨⊤⟩
 
+/-- Given a sieve `S` on `X : C`, its associated diagram `S.diagram` is defined to be
+    the natural functor from the full subcategory of the over category `C/X` consisting
+    of arrows in `S` to `C`. -/
+@[simp] def diagram (S : presieve X) : {f : over X // S f.hom} ⥤ C :=
```

Comment:
I'm not sure I like the `simp` tag here, and in the definition `cocone` below. Maybe using `abbreviation` would be better?

Reply:
Changed to `abbreviation`. `@[simp]` allows `dsimp` to unfold the definition during proof development. I imagine people using this PR would also need to unfold `cocone`, so I think using `@[simp] def` or `abbreviation` to make it easily unfoldable is a good idea.

Reply:
Another option is to use `@[simps]` to obtain the projections directly, which is usually a more useful approach when compared with `@[simp]` (since you obtain associated lemmas). I'm happy with `abbreviation` though. It shouldn't be too difficult to change later if needed.